apps/api/src/__tests__/snips/crawl.test.ts

```diff
@@ -179,4 +180,55 @@ describe("Crawl tests", () => {
             }
         }
     }, 5 * scrapeTimeout);
+
+    it.concurrent("rejects crawl when URL depth exceeds maxDepth", async () => {
+        const response = await request("http://127.0.0.1:3002")
+            .post("/v1/crawl")
+            .set("Authorization", `Bearer ${identity.apiKey}`)
+            .set("Content-Type", "application/json")
+            .send({
+                url: "https://firecrawl.dev/blog/category/deep/nested/path",
+                maxDepth: 2,
+                limit: 5,
+            });
+
+        expect(response.statusCode).toBe(400);
+        expect(response.body.success).toBe(false);
+        expect(response.body.error).toBe("Bad Request");
+        expect(response.body.details).toBeDefined();
+        expect(response.body.details[0].message).toBe("URL depth exceeds the specified maxDepth");
+        expect(response.body.details[0].path).toEqual(["url"]);
+    });
+
+    it.concurrent("accepts crawl when URL depth equals maxDepth", async () => {
+        const response = await request("http://127.0.0.1:3002")
+            .post("/v1/crawl")
+            .set("Authorization", `Bearer ${identity.apiKey}`)
+            .set("Content-Type", "application/json")
+            .send({
+                url: "https://firecrawl.dev/blog/category",
+                maxDepth: 2,
+                limit: 5,
+            });
+
+        expect(response.statusCode).toBe(200);
+        expect(response.body.success).toBe(true);
+        expect(typeof response.body.id).toBe("string");
+    });
+
+    it.concurrent("accepts crawl when URL depth is less than maxDepth", async () => {
+        const response = await request("http://127.0.0.1:3002")
+            .post("/v1/crawl")
+            .set("Authorization", `Bearer ${identity.apiKey}`)
+            .set("Content-Type", "application/json")
+            .send({
+                url: "https://firecrawl.dev/blog",
+                maxDepth: 5,
+                limit: 5,
+            });
+
+        expect(response.statusCode).toBe(200);
+        expect(response.body.success).toBe(true);
+        expect(typeof response.body.id).toBe("string");
+    });
```

Comment:
you're not allowed to use raw HTTP requests outside of lib.ts. move it there, called maybe smth like crawlStartRaw